File: data-model/src/main/java/com/medziku/plantwatering/station/parts/BatteryLevelSensor.java
package com.medziku.plantwatering.station.parts;

import com.medziku.plantwatering.station.AbstractPlantStationPart;


public abstract class BatteryLevelSensor extends AbstractPlantStationPart {

    protected float[] mBatteryLevels;

    public float[] getBatteryLevels() {
        return mBatteryLevels;
    }

    public void setBatteryLevels(float[] batteryLevels) {
        this.mBatteryLevels = batteryLevels;
    }

    public int getCellCount() {
        return mBatteryLevels.length;
    }
}

File: data-model/src/main/kotlin/com/medziku/plantwatering/station/parts/BatteryLevelSensor.kt
package com.medziku.plantwatering.station.parts

import com.medziku.plantwatering.station.PasivePlantStationPart


abstract class BatteryLevelSensor(override var name: String, var batteryLevels: FloatArray = floatArrayOf()) : PasivePlantStationPart {

    val cellCount: Int
        get() = batteryLevels.size
}
